### ISSUE TITLE
Undefined array key "slug" when generating docs with an empty groups.order config

### ISSUE THREAD

**Issue by bjhijmans** [knuckleswtf/scribe#545]
- [x] **I confirm that I have read and attempted the tips in the [Troubleshooting Guide](https://scribe.knuckles.wtf/laravel/troubleshooting).**

**What happened?**
1. I upgraded to the latest scribe version, and the config updater filled the 'order' field in the config
2. I added a new route and decided that maintaining the order array was a pain and to just use the default, so I set it to `[]`
3. I tried to build the docs, but I now get an error:

```
 Undefined array key "slug" (View: /home/vagrant/code/vendor/knuckleswtf/scribe/resources/views/themes/default/sidebar.blade.php)
```
on line 27.

When I add anything else, like `['']`, it works fine.

**My environment:**
 - PHP version 8.x 
 - Framework (Laravel/Lumen): Laravel
 - Laravel/Lumen version (from `composer show laravel/framework` or `composer show laravel/lumen-framework`): 9.36.3
 - Scribe version (from `composer show knuckleswtf/scribe`): 4.1.0

**My Scribe config (minus the comments):**

```php

    'theme' => 'default',
    'title' => config('app.name') . ' API Documentation',
    'description' => '',
    'base_url' => null,
    'routes' => [[
        'match' => [
            'domains' => ['*'],
            'prefixes' => ['api/*'],
            'versions' => ['v1'],
        ],
        'include' => [],
        'exclude' => [],
        'apply' => [
            'headers' => ['Content-Type' => 'application/json', 'Accept' => 'application/json'],
            'response_calls' => [
                'methods' => [],
                'config' => ['app.env' => 'documentation'],
                'queryParams' => [],
                'bodyParams' => [],
                'fileParams' => [],
                'cookies' => [],
            ],
        ],
    ]],
    'type' => 'laravel',
    'static' => [
        'output_path' => 'public/docs',
    ],
    'laravel' => [
        'add_routes' => false,
        'docs_url' => '/api-documentation',
        'middleware' => [],
        'assets_directory' => null,
    ],

    'try_it_out' => [
        'enabled' => false,
        'base_url' => null,
        'use_csrf' => false,
        'csrf_url' => '/sanctum/csrf-cookie',
    ],
    'auth' => [
        'enabled' => true,
        'default' => true,
        'in' => 'bearer',
        'name' => 'key',
        'use_value' => env('SCRIBE_AUTH_KEY'),
        'placeholder' => '{API_TOKEN}',
        'extra_info' => '',
    ],
    'intro_text' => <<<INTRO
 <snip>
INTRO

    ,
    'example_languages' => [
        'bash',
        'javascript',
        'python',
        'php',
    ],
    'postman' => [
        'enabled' => true,
        'overrides' => [
            // 'info.version' => '2.0.0',
        ],
    ],
    'openapi' => [
        'enabled' => true,
        'overrides' => [
            // 'info.version' => '2.0.0',
        ],
    ],
    'logo' => false,
    'strategies' => [
        'metadata' => [
            Strategies\Metadata\GetFromDocBlocks::class,
        ],
        'urlParameters' => [
//            Strategies\UrlParameters\GetFromLaravelAPI::class,
//            Strategies\UrlParameters\GetFromLumenAPI::class,
//            Strategies\UrlParameters\GetFromUrlParamTag::class,
            \App\Http\Api\Docs\Strategies\GetFromLaravelApiWithRandomUuid::class
        ],
        'queryParameters' => [
            Strategies\QueryParameters\GetFromFormRequest::class,
            Strategies\QueryParameters\GetFromInlineValidator::class,
            Strategies\QueryParameters\GetFromQueryParamTag::class,
            \App\Http\Api\Docs\Strategies\AddFilterParam::class,
            \App\Http\Api\Docs\Strategies\AddSortParam::class,
            \App\Http\Api\Docs\Strategies\AddIncludeParam::class,
        ],
        'headers' => [
            Strategies\Headers\GetFromRouteRules::class,
            Strategies\Headers\GetFromHeaderTag::class,
        ],
        'bodyParameters' => [
            Strategies\BodyParameters\GetFromFormRequest::class,
            Strategies\BodyParameters\GetFromInlineValidator::class,
            Strategies\BodyParameters\GetFromBodyParamTag::class,
            \App\Http\Api\Docs\Strategies\CheckFormRequestRules::class,
        ],
        'responses' => [
            Strategies\Responses\UseTransformerTags::class,
            Strategies\Responses\UseResponseTag::class,
            Strategies\Responses\UseResponseFileTag::class,
            Strategies\Responses\UseApiResourceTags::class,
            Strategies\Responses\ResponseCalls::class,
            \App\Http\Api\Docs\Strategies\AddExampleResponses::class,
            \App\Http\Api\Docs\Strategies\AddPaginatedExampleResponses::class,
        ],
        'responseFields' => [
            Strategies\ResponseFields\GetFromResponseFieldTag::class,
        ],
    ],

    'fractal' => [
        'serializer' => null,
    ],
    'routeMatcher' => \Knuckles\Scribe\Matching\RouteMatcher::class,
    'database_connections_to_transact' => [config('database.default'), config('database.default')],
    'groups' => [
        'default' => 'Endpoints',
        'order' => [],
    ],
    'last_updated' => 'Last updated: {date:F j, Y}',

    'examples' => [
        'faker_seed' => null,
        'models_source' => ['factoryCreate', 'factoryMake', 'databaseFirst'],
    ]
```

**Additional info:**

**Comment by shalvah** [comment]
Welp, thanks for reporting. I see you have a number of custom strategies. Was upgrading an issue?

**Comment by bjhijmans** [comment]
Nope, upgrading was awesome. I just had to run the upgrader and it told me how to fix my strategies.

**Comment by shalvah** [comment]
I can't reproduce this. Generating works fine with an empty order. Can you share more details? I need the stack trace, and ideally a dump of the `headings` array at that location.

**Comment by bjhijmans** [comment]
I took another look at what's causing it, and it's related to our append.md.

With `'order' => []` and the following append.md:

```markdown
# Other Objects

## Thing
```

It generates the following headings:

```php
[
 // lots of correct headings

 4 => [
    "slug" => "other-objects"
    "name" => "Other Objects"
    "subheadings" => []
  ]
  19 => [
    "subheadings" => [
      0 => [
        "slug" => "thing"
        "name" => "Thing"
        "subheadings" => []
      ]
    ]
  ]
]
```

When I change the order to anything else, like `'order' => ['']`, it generates the following headings

```php
[
 // lots of correct headings

 19 => [
    "slug" => "other-objects"
    "name" => "Other Objects"
    "subheadings" => [
      0 => [
        "slug" => "thing"
        "name" => "Thing"
        "subheadings" => []
      ]
    ]
  ]
```

As you can see, it doesn't put the h2 in the h1 as a subheading, putting it in a new h1 without a name or slug instead.

Stack trace:
```
[2022-11-09 10:18:49] local.ERROR: Undefined array key "slug" (View: /home/vagrant/code/vendor/knuckleswtf/scribe/resources/views/themes/default/sidebar.blade.php) (View: /home/vagrant/code/vendor/knuckleswtf/scribe/resources/views/themes/default/sidebar.blade.php) {"tags":{"app_url":"https://project.local"},"
exception":"[object] (Illuminate\\View\\ViewException(code: 0): Undefined array key \"slug\" (View: /home/vagrant/code/vendor/knuckleswtf/scribe/resources/views/themes/default/sidebar.blade.php) (View: /home/vagrant/code/vendor/knuckleswtf/scribe/resources/views/themes/default/sidebar.blade.php) at /home/vagrant/code/storage/framework/views/371da52989c5825af815b31e39e3516bcd18be2f.php:27)
[stacktrace]
#0 /home/vagrant/code/vendor/laravel/framework/src/Illuminate/View/Engines/PhpEngine.php(60): Illuminate\\View\\Engines\\CompilerEngine->handleViewException()
#1 /home/vagrant/code/vendor/laravel/framework/src/Illuminate/View/Engines/CompilerEngine.php(70): Illuminate\\View\\Engines\\PhpEngine->evaluatePath()
#2 /home/vagrant/code/vendor/laravel/framework/src/Illuminate/View/View.php(139): Illuminate\\View\\Engines\\CompilerEngine->get()
#3 /home/vagrant/code/vendor/laravel/framework/src/Illuminate/View/View.php(122): Illuminate\\View\\View->getContents()
#4 /home/vagrant/code/vendor/laravel/framework/src/Illuminate/View/View.php(91): Illuminate\\View\\View->renderContents()
#5 /home/vagrant/code/vendor/knuckleswtf/scribe/src/Writing/HtmlWriter.php(64): Illuminate\\View\\View->render()
#6 /home/vagrant/code/vendor/knuckleswtf/scribe/src/Writing/Writer.php(196): Knuckles\\Scribe\\Writing\\HtmlWriter->generate()
#7 /home/vagrant/code/vendor/knuckleswtf/scribe/src/Writing/Writer.php(70): Knuckles\\Scribe\\Writing\\Writer->writeHtmlDocs()
#8 /home/vagrant/code/vendor/knuckleswtf/scribe/src/Commands/GenerateDocumentation.php(69): Knuckles\\Scribe\\Writing\\Writer->writeDocs()
#9 /home/vagrant/code/vendor/laravel/framework/src/Illuminate/Container/BoundMethod.php(36): Knuckles\\Scribe\\Commands\\GenerateDocumentation->handle()
#10 /home/vagrant/code/vendor/laravel/framework/src/Illuminate/Container/Util.php(41): Illuminate\\Container\\BoundMethod::Illuminate\\Container\\{closure}()
#11 /home/vagrant/code/vendor/laravel/framework/src/Illuminate/Container/BoundMethod.php(93): Illuminate\\Container\\Util::unwrapIfClosure()
#12 /home/vagrant/code/vendor/laravel/framework/src/Illuminate/Container/BoundMethod.php(37): Illuminate\\Container\\BoundMethod::callBoundMethod()
#13 /home/vagrant/code/vendor/laravel/framework/src/Illuminate/Container/Container.php(651): Illuminate\\Container\\BoundMethod::call()
#14 /home/vagrant/code/vendor/laravel/framework/src/Illuminate/Console/Command.php(144): Illuminate\\Container\\Container->call()
#15 /home/vagrant/code/vendor/symfony/console/Command/Command.php(291): Illuminate\\Console\\Command->execute()
#16 /home/vagrant/code/vendor/laravel/framework/src/Illuminate/Console/Command.php(126): Symfony\\Component\\Console\\Command\\Command->run()
#17 /home/vagrant/code/vendor/symfony/console/Application.php(1002): Illuminate\\Console\\Command->run()
#18 /home/vagrant/code/vendor/symfony/console/Application.php(299): Symfony\\Component\\Console\\Application->doRunCommand()
#19 /home/vagrant/code/vendor/symfony/console/Application.php(171): Symfony\\Component\\Console\\Application->doRun()
#20 /home/vagrant/code/vendor/laravel/framework/src/Illuminate/Console/Application.php(102): Symfony\\Component\\Console\\Application->run()
#21 /home/vagrant/code/vendor/laravel/framework/src/Illuminate/Foundation/Console/Kernel.php(151): Illuminate\\Console\\Application->run()
#22 /home/vagrant/code/artisan(35): Illuminate\\Foundation\\Console\\Kernel->handle()
#23 {main}

```

**Comment by shalvah** [comment]
Fixed in 4.2.1
